File: P2/Ejercicio1_MantenimientoFlutter/lib/filters/PasswordNumberFilter.dart
import '../interfaces/Filter.dart';

// filtro que comprueba que la contraseña tenga al menos un número
class PasswordNumberFilter implements Filter {
  @override
  String? execute(Map<String, String> request) {
    // Asegúrate de que la llave es exactamente 'password'
    final password = request['password'] ?? '';
    
    // hasMatch devolverá true si encuentra al menos un número del 0 al 9
    if (!RegExp(r'[0-9]').hasMatch(password)) {
      return "La contraseña debe contener al menos un número";
    }
    
    return null;
  }
}

File: P1/Ejercicio4/filters/password_number_filter.rb
# filtro que valida que la contraseña contenga al menos un número
# implementa la interfaz Filter
class PasswordNumberFilter
  include Filter

  def execute(request)
    # comprueba si la contraseña contiene algún dígito (0-9)
    # mediante una expresión regular
    unless request[:password].match(/\d/)
      raise ValidationError, "Debe contener un número"
    end
  end
end